### DISCUSSION TITLE
Help！ SIX1268 fail connect to ESP32-S2 .

### DISCUSSION THREAD

**Discussion by skylai82** [sandeepmistry/arduino-LoRa#678] [Q&A]
```
#include <SPI.h>
#define SCK 36  // SIX1268 SCK connect to ESP32-S2  GPIO 36
#define MOSI 37 // SIX1268 MOSI connect to ESP32-S2 GPIO 37
#define MISO 35 // SIX1268 MISO connect to ESP32-S2 GPIO 35
#define SS  38  // SIX1268 NSS connect to ESP32-S2 GPIO 38

#include <LoRa.h>
#define RST 10  // SIX1268 Reset connect to ESP32-S2 的GPIO 10
#define DI01 14  // SIX1268 DIO1 connect to ESP32-S2 的GPIO 14
#define BUSY 13  // SIX1268 DIO1 connect to ESP32-S2 的GPIO 13

void setup() {
  Serial.begin(115200);
  delay(1000);
    // SPI pin set
  SPI.begin(SCK, MISO, MOSI, SS);
  Serial.println("Set SPI pin");
  delay(1000);
  // Initialize LoRa
  LoRa.setPins(SS, RST, DI0);//, BUSY_PIN for what?
  Serial.println("Set Lora pin");
  delay(1000);

  Serial.println("LoRa Receiver");

  if (!LoRa.begin(433E6)) {
    Serial.println("Starting LoRa failed!");
    //while (1);
  }
}

void loop() {
  // try to parse packet
  int packetSize = LoRa.parsePacket();
  if (packetSize) {
    // received a packet
    Serial.print("Received packet '");

    // read packet
    while (LoRa.available()) {
      Serial.print((char)LoRa.read());
    }

    // print RSSI of packet
    Serial.print("' with RSSI ");
    Serial.println(LoRa.packetRssi());
  }
}
```
Why connection fail???

**Comment by morganrallen** [comment]
You may need to create a new SPI interface instead of changing the pins on the default one, though I'm not 100% sure how that all work.  Instead of `SPI.begin...` try...

```
// create new SPI interface
SPIClass SPI_LORA(HSPI);
SPI_LORA.begin(SCK, MISO, MOSI, SS);

// attach SPI interface to LoRa
LoRa.setSPI(SPI_LORA);
```

**Reply by skylai82** [reply]
/*
 * ESP32-S2
 * Lora SIX1268
 * 
 */

#include <SPI.h>
#define SCK 36  // SIX1268 SCK connect to ESP32-S2 的GPIO 36
#define MOSI 37 // SIX1268 MOSI connect to ESP32-S2 的GPIO 37
#define MISO 35 // SIX1268 MISO connect to ESP32-S2 的GPIO 35
#define SS  38  // SIX1268 NSS connect to ESP32-S2 的GPIO 38

#include <LoRa.h>
#define RST 10  // SIX1268 Reset connect to ESP32-S2 的GPIO 10
#define DI0 14  // SIX1268 DIO1 connect to ESP32-S2 的GPIO 14
#define BUSY 13  // SIX1268 DIO1 connect to ESP32-S2 的GPIO 13

void setup() {

  Serial.begin(115200);
  delay(2000);
  Serial.println("");
  Serial.println("LoRa Power On");

  // create new SPI interface
  Serial.println("create new SPI interface");
  SPIClass SPI_LORA(HSPI);
  SPI_LORA.begin(SCK, MISO, MOSI, SS);
  delay(2000);
  
  // attach SPI interface to LoRa
  LoRa.setSPI(SPI_LORA);

  Serial.println("LoRa initialization...");

  if (!LoRa.begin(433E6)) {
    Serial.println("LoRa initialization failed. Check your connections.");
    while (1);
  }
  Serial.println("LoRa receiver started");
}

void loop() {

}

![image](https://github.com/sandeepmistry/arduino-LoRa/assets/37041202/ddb1e468-df3d-4404-a926-962a61abf43a)

**Tried, still failed**
